### PR TITLE
Feature/add data resolvers

### PR DESCRIPTION
**Pull request by MartinLarroca**
Done:
    - Resolved graphQL n + 1 problem in all micro services through batching, for achieving this, a library
       called DataLoader ,which uses batching to solve this problem, was introduced to de project.
    - Also improved code quality in different parts of the project.

### CODE REVIEW

File: twitterFederation/services/comments/src/dataloaders/comment.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  const comments = await Comment.findAll({ where: { id: ids } });
+  const groupedComments = groupBy((elem) => elem.tweetId.toString(), comments);
+  return ids.map((id) => ({
+    __typename: 'User',
+    id: groupedComments[id][0].userId,
+  }));
+};
+
+const commentsTweetReference = async (ids: string[]) => {
+  const comments = await Comment.findAll({ where: { id: ids } });
```
Comment (emidesejas) [inline twitterFederation/services/comments/src/dataloaders/comment.ts:38]: Maybe select the `tweetId` directly here

File: twitterFederation/services/comments/src/dataloaders/comment.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  const groupedComments = groupBy((elem) => elem.tweetId.toString(), comments);
+  return ids.map((id) => ({
+    __typename: 'Tweet',
+    id: groupedComments[id][0].tweetId,
```
Comment (emidesejas) [inline twitterFederation/services/comments/src/dataloaders/comment.ts:42]: Also I think the id is the Comment Id and it's grouped by tweet ids

File: twitterFederation/services/comments/src/dataloaders/comment.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  }));
+};
+
+const create = (value: DataLoaderType) => {
```
Comment (emidesejas) [inline twitterFederation/services/comments/src/dataloaders/comment.ts:46]: Nice

**Review by emidesejas — APPROVED** [review]
Nice work! review the comments I left and also the context type that I mentioned you through slack

**Review by jpbalarini — APPROVED** [review]
Nice work!

**Pull request merged**